<!DOCTYPE html>
<html><head><link type="text/css" rel="stylesheet" href="/tracker/style.css"> </link><script type="text/javascript" src="/tracker/script.js"> </script><title>CVE-2018-1000802</title></head><body><header><h1>CVE-2018-1000802</h1></header><table><tr><td><b>Name</b></td><td>CVE-2018-1000802</td></tr><tr><td><b>Description</b></td><td>Python Software Foundation Python (CPython) version 2.7 contains a CWE-77: Improper Neutralization of Special Elements used in a Command ('Command Injection') vulnerability in shutil module (make_archive function) that can result in Denial of service, Information gain via injection of arbitrary files on the system or entire drive. This attack appear to be exploitable via Passage of unfiltered user input to the function. This vulnerability appears to have been fixed in after commit add531a1e55b0a739b0f42582f1c9747e5649ace.</td></tr><tr><td><b>Source</b></td><td><a href="https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2018-1000802">CVE</a> (at <a href="https://nvd.nist.gov/vuln/detail/CVE-2018-1000802">NVD</a>; <a href="https://www.kb.cert.org/vuls/byid?searchview=&amp;query=CVE-2018-1000802">CERT</a>, <a href="https://lwn.net/Search/DoSearch?words=CVE-2018-1000802">LWN</a>, <a href="https://marc.info/?l=oss-security&amp;s=CVE-2018-1000802">oss-sec</a>, <a href="https://marc.info/?l=full-disclosure&amp;s=CVE-2018-1000802">fulldisc</a>, <a href="https://marc.info/?l=bugtraq&amp;s=CVE-2018-1000802">bugtraq</a>, <a href="https://www.exploit-db.com/search/?action=search&amp;cve=2018-1000802">EDB</a>, <a href="https://www.rapid7.com/db/search?q=CVE-2018-1000802">Metasploit</a>, <a href="https://bugzilla.redhat.com/show_bug.cgi?id=CVE-2018-1000802">Red Hat</a>, <a href="https://people.canonical.com/~ubuntu-security/cve/CVE-2018-1000802">Ubuntu</a>, <a href="https://bugs.gentoo.org/show_bug.cgi?id=CVE-2018-1000802">Gentoo</a>, SUSE <a href="https://bugzilla.suse.com/show_bug.cgi?id=CVE-2018-1000802">bugzilla</a>/<a href="https://www.suse.com/security/cve/CVE-2018-1000802/">CVE</a>, <a href="https://advisories.mageia.org/CVE-2018-1000802.html">Mageia</a>, GitHub <a href="https://github.com/search?type=Code&amp;q=%22CVE-2018-1000802%22">code</a>/<a href="https://github.com/search?type=Issues&amp;q=%22CVE-2018-1000802%22">issues</a>, <a href="https://duckduckgo.com/html?q=%22CVE-2018-1000802%22">web search</a>, <a href="https://oss-security.openwall.org/wiki/vendors">more</a>)</td></tr><tr><td><b>References</b></td><td><a href="/tracker/DLA-1519-1">DLA-1519-1</a>, <a href="/tracker/DLA-1520-1">DLA-1520-1</a>, <a href="/tracker/DSA-4306-1">DSA-4306-1</a></td></tr><tr><td><b>NVD severity</b></td><td>high</td></tr><tr><td><b>Debian Bugs</b></td><td><a href="https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=909673">909673</a></td></tr></table><h2>Vulnerable and fixed packages</h2><p>The table below lists information on source packages.</p><table><tr><th>Source Package</th><th>Release</th><th>Version</th><th>Status</th></tr><tr><td><a href="/tracker/source-package/python2.7">python2.7</a> (<a href="https://tracker.debian.org/pkg/python2.7">PTS</a>)</td><td>stretch</td><td>2.7.13-2+deb9u3</td><td>fixed</td></tr><tr><td></td><td>stretch (security)</td><td>2.7.13-2+deb9u5</td><td>fixed</td></tr><tr><td></td><td>buster</td><td>2.7.16-2+deb10u1</td><td>fixed</td></tr><tr><td></td><td>bullseye, sid</td><td>2.7.18-7</td><td>fixed</td></tr><tr><td><a href="/tracker/source-package/python3.5">python3.5</a> (<a href="https://tracker.debian.org/pkg/python3.5">PTS</a>)</td><td>stretch</td><td>3.5.3-1+deb9u1</td><td>fixed</td></tr><tr><td></td><td>stretch (security)</td><td>3.5.3-1+deb9u4</td><td>fixed</td></tr><tr><td><a href="/tracker/source-package/python3.7">python3.7</a> (<a href="https://tracker.debian.org/pkg/python3.7">PTS</a>)</td><td>buster</td><td>3.7.3-2+deb10u3</td><td>fixed</td></tr></table><p>The information below is based on the following data on fixed versions.</p><table><tr><th>Package</th><th>Type</th><th>Release</th><th>Fixed Version</th><th>Urgency</th><th>Origin</th><th>Debian Bugs</th></tr><tr><td><a href="/tracker/source-package/python2.7">python2.7</a></td><td>source</td><td>jessie</td><td>2.7.9-2+deb8u2</td><td></td><td><a href="/tracker/DLA-1519-1">DLA-1519-1</a></td><td></td></tr><tr><td><a href="/tracker/source-package/python2.7">python2.7</a></td><td>source</td><td>stretch</td><td>2.7.13-2+deb9u3</td><td></td><td><a href="/tracker/DSA-4306-1">DSA-4306-1</a></td><td></td></tr><tr><td><a href="/tracker/source-package/python2.7">python2.7</a></td><td>source</td><td>(unstable)</td><td>2.7.15-5</td><td></td><td></td><td><a href="https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=909673">909673</a></td></tr><tr><td><a href="/tracker/source-package/python3.4">python3.4</a></td><td>source</td><td>jessie</td><td>3.4.2-1+deb8u1</td><td></td><td><a href="/tracker/DLA-1520-1">DLA-1520-1</a></td><td></td></tr><tr><td><a href="/tracker/source-package/python3.4">python3.4</a></td><td>source</td><td>(unstable)</td><td><span class="red">(unfixed)</span></td><td></td><td></td><td></td></tr><tr><td><a href="/tracker/source-package/python3.5">python3.5</a></td><td>source</td><td>(unstable)</td><td>(not affected)</td><td></td><td></td><td></td></tr><tr><td><a href="/tracker/source-package/python3.6">python3.6</a></td><td>source</td><td>(unstable)</td><td>(not affected)</td><td></td><td></td><td></td></tr><tr><td><a href="/tracker/source-package/python3.7">python3.7</a></td><td>source</td><td>(unstable)</td><td>(not affected)</td><td></td><td></td><td></td></tr></table><h2>Notes</h2><pre><SPAN>- python3.7 &lt;not-affected&gt; (Fixed before initial upload)</SPAN><br/><SPAN>- python3.6 &lt;not-affected&gt; (Fixed before initial upload)</SPAN><br/><SPAN>- python3.5 &lt;not-affected&gt; (Fixed before initial upload)</SPAN><br/><a href="https://bugs.python.org/issue34540">https://bugs.python.org/issue34540</a><br/><a href="https://github.com/python/cpython/commit/d8b103b8b3ef9644805341216963a64098642435">https://github.com/python/cpython/commit/d8b103b8b3ef9644805341216963a64098642435</a><br/><SPAN>Later versions did remove _call_external_zip with</SPAN><br/><a href="https://github.com/python/cpython/commit/a0934b2c1b939fdebee8dc18d49a0f6c52324773">https://github.com/python/cpython/commit/a0934b2c1b939fdebee8dc18d49a0f6c52324773</a><br/><SPAN>which used distutils.spawn.</SPAN><br/>PoC: <a href="https://mega.nz/#!JUFiCC4R!mq-jQ8ySFwIhX6WMDujaZuNBfttDVt7DETlfOIQE1ig">https://mega.nz/#!JUFiCC4R!mq-jQ8ySFwIhX6WMDujaZuNBfttDVt7DETlfOIQE1ig</a><br/></pre><hr/><form method="get" id="searchform" action="/tracker/">Search for package or bug name: <input type="text" name="query" onkeyup="onSearch(this.value)" onmousemove="onSearch(this.value)"/><input type="submit" value="Go"/> <a href="/tracker/data/report">Reporting problems</a></form><footer><p><a href="/tracker/">Home</a> - <a href="https://www.debian.org/security/">Debian Security</a> - <a href="https://salsa.debian.org/security-tracker-team/security-tracker/blob/master/bin/tracker_service.py">Source</a> <a href="https://salsa.debian.org/security-tracker-team/security-tracker">(Git)</a></p></footer></body></html>